pandas/tests/plotting/test_frame.py

```diff
@@ -1190,6 +1190,16 @@ def test_scatterplot_object_data(self):
         _check_plot_works(df.plot.scatter, x="a", y="b")
         _check_plot_works(df.plot.scatter, x=0, y=1)
 
+    def test_scatterplot_color_by_categorical(self):
+        df = pd.DataFrame(
+            [[5.1, 3.5], [4.9, 3.0], [7.0, 3.2], [6.4, 3.2], [5.9, 3.0]],
+            columns=["length", "width"],
+        )
+        df["species"] = pd.Categorical(
+            ["setosa", "setosa", "virginica", "virginica", "versicolor"]
+        )
+        _check_plot_works(df.plot.scatter, x=0, y=1, c="species")
```

Comment:
Is there way within the test to check the legend is drawn and the appropriate colors are applied?